rclcpp_lifecycle/src/lifecycle_node_entities_manager.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "lifecycle_node_entities_manager.hpp"
+
+namespace rclcpp_lifecycle
+{
+
+void
+LifecycleNodeEntitiesManager::on_activate() const
```

Comment:
Is this separate class for managing entities needed?

Reply:
No, just seemed odd for the `LifecycleNodeInterfaceImpl` to manage these entities internally but delegate all other responsibilities outward. This felt* cleaner / easier to switch out later but I don't have a preference toward keeping it.

Reply:
Staying with the separation as I think it is a slightly easier story to sell / easy enough to copy / paste back